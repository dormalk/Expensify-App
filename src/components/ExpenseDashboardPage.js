import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummey from './ExpensesSummery';
const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummey/>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
