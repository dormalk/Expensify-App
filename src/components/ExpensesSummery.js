import React from 'react';
import { connect } from 'react-redux';
import expenses from '../selectors/expenses';
import selectedExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummery = ({expensesCount,expensesTotal}) => {
    const expensesWord = expensesCount === 1 ? 'expense' : 'expenses';
    const expensesTotalNumeral = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div>
            <h2>Viewing {expensesCount} {expensesWord} totalling {expensesTotalNumeral}</h2>
        </div>
    );
};
const mapStateToProps = (state) =>{
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectedExpensesTotal(visibleExpenses)    
    };
};

export default connect(mapStateToProps)(ExpensesSummery);