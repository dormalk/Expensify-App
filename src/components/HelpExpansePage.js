import React from 'react';

const HelpExpansePage = () => (
    <div>
       This is help page
    </div>
);

export default HelpExpansePage;