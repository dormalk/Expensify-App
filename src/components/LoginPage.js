import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-login">
    <div className="box-login__box">
      <h2 className="box-login__title">Expensify App</h2>
      <p>It's time tou controll on your expenses</p>
      <button className="button" onClick={startLogin}>Login with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
