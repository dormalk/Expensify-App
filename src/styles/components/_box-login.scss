.box-login{
    align-items: center;
    background-image: url('/images/background_login.jpg');
    background-size:cover;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;
}

.box-login__box{
    background: fade-out(white, .15);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width:25rem;
}

.box-login__title{
    margin: 0 0 $m-size 0;
    line-height: 1;

}