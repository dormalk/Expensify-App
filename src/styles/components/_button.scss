.button{
    background-color: #d83e3e;
    border-radius: 2px;
    border: 0px;
    padding: $xs-size $xl-size; 
    color: white;
    font-size: $m-size;
    border-bottom: 3px solid #b72d2d;
}


.button:active{
    background-color: #cc3b3b;
}


.button__add{
    background-color: #3e83d8;
    color:white;
    padding: $s-size $m-size; 
    text-decoration: none;
    box-shadow: 1px 1px 3px #c4c4c4;
    display: inline-block;
    border: 0px;
}

.button__add:active{
    background-color: #3a7ccc;
}

.button__logout{
    background: none;
    color:white;
    border: 0px;
}

.button__remove{
    background-color: #a0a0a0;
    color:white;
    padding: $s-size $m-size; 
    text-decoration: none;
    box-shadow: 1px 1px 3px #c4c4c4;
    display: inline-block;
    border: 0px;
}

.button__remove:active{
    background-color: #8c8a8a;
}
