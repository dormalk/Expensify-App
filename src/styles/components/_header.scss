.header{
    background: $dark-blue;
}

.header__title{
    color:white;
    text-decoration: none;
    h1{
        margin: 0px;
    }
}

.header__content{
    align-items: center;
    display: flex;
    justify-content: space-between;
    padding: $s-size 0;
}