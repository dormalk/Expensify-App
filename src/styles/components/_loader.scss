.loader{
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    width: 100vw;
}

.loader__image{
    height: 10rem;
    width: 10rem;
}