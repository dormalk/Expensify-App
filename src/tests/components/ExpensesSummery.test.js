import React from 'react';
import { shallow } from 'enzyme';
import {ExpensesSummery} from '../../components/ExpensesSummery';

test('should render correctly ExpensesSummery with 1 expense',() => {
    const wrapper = shallow(<ExpensesSummery expensesCount={1} expensesTotal={123}/>);
    expect(wrapper).toMatchSnapshot();
});


test('should render correctly ExpensesSummery with multipale expenses',() => {
    const wrapper = shallow(<ExpensesSummery expensesCount={26} expensesTotal={19913233123}/>);
    expect(wrapper).toMatchSnapshot();
});