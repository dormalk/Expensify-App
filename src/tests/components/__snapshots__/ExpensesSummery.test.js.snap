// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly ExpensesSummery with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $1.23
  </h2>
</div>
`;

exports[`should render correctly ExpensesSummery with multipale expenses 1`] = `
<div>
  <h2>
    Viewing 
    26
     
    expenses
     totalling 
    $199,132,331.23
  </h2>
</div>
`;
